website/templates/project/view_file.mako

```diff
@@ -46,17 +51,183 @@
     </div>
 
     <div class="panel-expand col-md-6">
-
         <div id="fileRendered" class="mfr mfr-file">
-          % if rendered is not None:
-            ${rendered}
-          % else:
-            <img src="/static/img/loading.gif">
-          % endif
+            <div class="wiki" id="filePageContext">
+
+            % if user['can_edit'] and is_editable:
+
+                <div data-bind="with: $root.editVM.fileEditor.viewModel" data-osf-panel="Edit" style="display: none">
+                    <div class="osf-panel" >
+                        <div class="osf-panel-header" >
+                            <div class="wiki-panel">
+                                <div class="wiki-panel-header">
+                                    <div class="row">
+                                        <div class="col-md-6">
+                                            <span class="wiki-panel-title" > <i class="fa fa-pencil-square-o"></i>   Edit </span>
+                                        </div>
+                                        <div class="col-md-6">
+                                            <div class="pull-right">
+                                                <div class="progress progress-no-margin pointer " data-toggle="modal" data-bind="attr: {data-target: modalTarget}" >
+                                                    <div role="progressbar" data-bind="attr: progressBar">
+                                                        <span class="progress-bar-content">
+                                                            <span data-bind="text: statusDisplay"></span>
+                                                            <span class="sharejs-info-btn">
+                                                                <i class="fa fa-question-circle fa-large"></i>
+                                                            </span>
+                                                        </span>
+                                                    </div>
+                                                </div>
+                                            </div>
+                                        </div>
+                                    </div>
+                                </div>
+                            </div>
+                        </div>
+
+                        <form id="file-edit-form">
+                            <div class="wiki-panel-body" style="padding: 10px">
+                                <div class="row">
+                                    <div class="col-xs-12">
+                                        <div class="form-group wmd-panel">
+                                            <ul class="list-inline" data-bind="foreach: activeUsers" class="pull-right">
+                                              <!-- ko ifnot: id === '${user_id}' -->
```

Comment:
My philosophical inclination is to use an explicit data-bind rather than a <!-- ko tag, but this is just an opinion.